File: robotReturn.py
"""Robot Return to origin"""

def judgeCircle(moves):
    """Return True if robot returns to origin"""
    y = 0
    x = 0
    for i in moves:
        if i == 'U':
            y += 1
        elif i == 'D':
            y -= 1
        elif i == 'L':
            x -= 1
        elif i == 'R':
            x += 1
    if x == 0 and y == 0:
        return True
    return False
    
    
print(judgeCircle("UDLURL"))

File: RobotReturn.java


public class RobotReturn {
   public static void main(String[] args) {
         String moves = "UD";
         System.out.println(judgeCircle(moves));
   }
   
   public static boolean judgeCircle(String str){
         int x = 0;
         int y = 0;
         for (int i = 0; i < str.length(); i++){
              if (str.charAt(i) == 'U'){
                y++;
              }
              else if (str.charAt(i) == 'D'){
                y--;
              }
              else if (str.charAt(i) == 'L'){
                x--;
              }
              else if (str.charAt(i) == 'R'){
                x++;
              }
         }
         return x == 0 && y == 0;
   }
}
